proxy/http/HttpTransactCache.cc

```diff
@@ -1309,7 +1309,7 @@ HttpTransactCache::match_response_to_request_conditionals(HTTPHdr *request, HTTP
 
   // If-Match: must match strongly //
   if (request->presence(MIME_PRESENCE_IF_MATCH)) {
-    int raw_etags_len, comma_sep_tag_list_len;
+    int raw_etags_len, comma_sep_tag_list_len = 0;
```

Comment:
I think `raw_etags_len` is still uninitialized. I usually don't define multiple variables on one line to avoid this confusion.
```c++
#include <iostream>
int main() {
    int a, b = 1;
    std::cout << a << std::endl;
    std::cout << b << std::endl;
}
```

```
foo.cc:4:18: warning: variable 'a' is uninitialized when used here [-Wuninitialized]
    std::cout << a << std::endl;
                 ^
foo.cc:3:10: note: initialize the variable 'a' to silence this warning
    int a, b = 1;
         ^
          = 0
```

Reply:
I think this is a different case, since the value is presumably set by value_get() in the next line.

Some advocate the practice of always initializing stack variables, to increase the repeatability of code behavior when debugging.  We could consider defining and using a TS utility macro like:
```
#ifdef DEBUG
#define TS_DEFDZ(TYPE, NAME) TYPE NAME = 0
#else
#define TS_DEFDZ(TYPE, NAME) TYPE NAME
#endif
```
I wonder if gcc/clang has a compile option that will cause code to be generated to zero out stack allocations, that we could apply when configure --enable-debug was used.

Reply:
> I think this is a different case, since the value is presumably set by value_get() in the next line.

Oh, I missed it. However, that proved this line is confusing. The reason I wrote the example code is I didn't remember the behavior and needed to check it. If I forgot the behavior, I might add a new variable on this line and forget initializing it.

Reply:
FTR, It was `comma_sep_tag_list_len` Xcode was warning on.

<img width="1269" alt="Screen Shot 2020-08-12 at 9 47 28 AM" src="https://user-images.githubusercontent.com/307125/90043371-decb2000-dc80-11ea-8ef8-90aa5be0ddd8.png">

Reply:
I agreed on what @maskit suggested.
1. Let's separate the declarations one per line.
2.  and initialize each of them.
That way is much clearer than the current code.